app/helpers/common/transformApiValidationResponse.js

```diff
@@ -0,0 +1,22 @@
+const transformPropertyName = propertyName => (propertyName.startsWith('[')
+  ? propertyName.substring(4)
+  : propertyName);
+
+export const transformApiValidationResponse = (validationResponse) => {
+  const distinctApiErrors = Object
```

Comment:
can the lambdas be jojned to save making an object that's just going to be returned?

Reply:
Do you mean something like this?

```javascript
export const transformApiValidationResponse = validationResponse => Object.entries(Object
  .entries(validationResponse)
  .reduce((acc, [key, value]) => {
    const field = transformPropertyName(key);

    if (acc[field] === undefined) {
      acc[field] = new Set();
    }

    value.forEach(v => acc[field].add(v));
    return acc;
  }, {}))
  .flatMap(([key, value]) => [...value].map(v => ({ field: key, id: v })));
```
I think the existing version is more readable and I'm not sure we're really gaining much by joining the two expressions. My preference would be to stick with the more readable version.

Reply:
yeah did mean that
although was going to suggest then swapping to a foreach or something for the entries bit
but yeah is more readable how you had it